Framework/PythonInterface/test/python/plugins/algorithms/HB2AReduceTest.py

```diff
@@ -71,14 +79,28 @@ def test_ExcludeDetectors(self):
     def test_anode_vs_temp(self):
         HB2AReduce_ws = HB2AReduce('HB2A_exp0660_scan0146.dat',
                                    Vanadium='HB2A_exp0644_scan0018.dat',
-                                   IndividualDetectors=True)
+                                   IndividualDetectors=True,
+                                   SaveData=False)
         self.assertTrue(HB2AReduce_ws)
         self.assertEqual(HB2AReduce_ws.getNumberHistograms(), 44)
         self.assertEqual(HB2AReduce_ws.blocksize(), 56)
         self.assertEqual(np.argmax(HB2AReduce_ws.readY(7)), 2)
         self.assertAlmostEquals(np.max(HB2AReduce_ws.readY(7)), 2.13258433)
         HB2AReduce_ws.delete()
 
+    def test_saving_files(self):
+        HB2AReduce_ws = HB2AReduce(
+            'HB2A_exp0660_scan0146.dat',
+            Vanadium='HB2A_exp0644_scan0018.dat',
+            IndividualDetectors=True,
+            OutputWorkspace="HB2AReduce_ws",
+            OutputFormat='XYE',
+            OutputDirectory=self._default_save_directory,
+        )
+        self.assertTrue(HB2AReduce_ws)
+        # the saved file cannot be reliable located in the build system
```

Comment:
I don't see a reason this wouldn't work. I think the issue here is using `tempfile`, maybe try using the location defined by `defaultsave.directory` in the mantid config like other tests.

Reply:
I tried out `defaultsave.directory`, and it turns out the default saving directory is an empty string, which means we are back to the original issue where the default saving folder is not present on the build server.

Reply:
here is the link to the commit where I have to switch to tempfile

https://github.com/mantidproject/mantid/pull/29951/commits/831294217caf879b4b93d81e3639d1952b8f37ec

btw, the majority of the dev&testing is done in the sibling branch that uses `ornl-next` as base.

Reply:
The issue is if I run the test it creates the file `/tmp/None.dat` not `/tmp/HB2AReduce_ws.dat`